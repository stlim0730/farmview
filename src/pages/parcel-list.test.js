import React from 'react';
import ParcelList from './parcel-list';
import renderer from 'react-test-renderer';
import jquery from 'jquery';
import { List } from 'immutable';

test('getSql returns SELECT * FROM central_coast_joined', () => {
		expect(ParcelList._getSql()).toBe('SELECT * FROM central_coast_joined');
});

test('rendered component matches snapshot', () => {
	const list = List([{acres: "1"}]);
	jest.mock('jquery', () => ({
		getJSON(url, callback) {
			expect(url).toBe("https://calo1.cartodb.com/api/v2/sql/?q=");
			var res = {rows: [{acres: 1}]};
			callback(res);
		}
	}));
	const parcelList = renderer.create(
    <ParcelList cartoUserName="calo1"></ParcelList>
  );
  expect(parcelList).toMatchSnapshot();
});