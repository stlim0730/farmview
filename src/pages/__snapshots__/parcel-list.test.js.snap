// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered component matches snapshot 1`] = `
<div
  className="ReactVirtualized__Table"
  role="grid"
  style={Object {}}
>
  <div
    className="ReactVirtualized__Table__headerRow"
    role="row"
    style={
      Object {
        "height": 30,
        "overflow": "hidden",
        "paddingRight": 0,
        "width": 500,
      }
    }
  >
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 80px",
          "flex": "0 1 80px",
          "msFlex": "0 1 80px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Index"
      >
        Index
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 80px",
          "flex": "0 1 80px",
          "msFlex": "0 1 80px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Size"
      >
        Size
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 80px",
          "flex": "0 1 80px",
          "msFlex": "0 1 80px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Location"
      >
        Location
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 80px",
          "flex": "0 1 80px",
          "msFlex": "0 1 80px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Zoning Type"
      >
        Zoning Type
      </span>
    </div>
    <div
      className="ReactVirtualized__Table__headerColumn"
      role="columnheader"
      style={
        Object {
          "WebkitFlex": "0 1 80px",
          "flex": "0 1 80px",
          "msFlex": "0 1 80px",
        }
      }
    >
      <span
        className="ReactVirtualized__Table__headerTruncatedText"
        title="Soil Type"
      >
        Soil Type
      </span>
    </div>
  </div>
  <div
    aria-label="grid"
    aria-readonly={true}
    className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
    onScroll={[Function]}
    role="rowgroup"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "boxSizing": "border-box",
        "direction": "ltr",
        "height": 370,
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "relative",
        "width": 500,
        "willChange": "transform",
      }
    }
    tabIndex={0}
  />
</div>
`;
